src/services/web3/connectors/metamask/metamask.connector.ts

```diff
@@ -0,0 +1,40 @@
+import { Connector } from '../connector';
+export class MetamaskConnector extends Connector {
+  id = 'injected';
+  async connect() {
+    // type for window.ethereum is causing conflicts (provided by some library)
+    const provider = window.ethereum as any;
+    if (provider) {
+      this.provider = provider;
+      this.active.value = true;
+
+      try {
+        const accounts = await provider.request({
+          method: 'eth_requestAccounts'
+        });
+
+        const chainId = await provider.request({ method: 'eth_chainId' });
```

Comment:
Seems like `accounts` and `chainId` can be combined to a `promise.all`

Reply:
I could do this, but I prefer the readability of having them separate